lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```diff
@@ -41,7 +41,7 @@ module.exports = {
 
     _.forEach(this.serverless.service.functions, (functionObject, functionName) => {
       (functionObject.events || []).forEach(event => {
-        if (_.has(event, 'http')) {
+        if ('http' in event) {
```

Comment:
Let's not use `in`, this should be simply `if (event.http)`

Reply:
Kind of similar to `lib/plugins/aws/provider/awsProvider.js`.
In `events/apiGateway/lib/validate.test.js`, `http` is set to `null` so `if (event.http)` will be false and it will not reach `this.getHttpPath(http, functionName)` which throws the error.

Reply:
It's confusingly constructed test. Lest' replace `null` with `{}`